Source/misdat.h

```diff
@@ -121,19 +122,57 @@ typedef struct MissileData {
 	MissileMovementDistrubution MovementDistribution;
 } MissileDataStruct;
 
-typedef struct MisFileData {
-	const char *mName;
-	uint8_t mAnimName;
-	uint8_t mAnimFAmt;
-	uint32_t mFlags;
-	byte *mAnimData[16];
-	uint8_t mAnimDelay[16];
-	uint8_t mAnimLen[16];
-	int16_t mAnimWidth[16];
-	int16_t mAnimWidth2[16];
-} MisFileDataStruct;
+struct MisFileData {
+	const char *name;
+	uint8_t animName;
+	uint8_t animFAmt;
+	uint32_t flags;
+	std::array<byte *, 16> animData = {};
+	std::array<uint8_t, 16> animDelay = {};
+	std::array<uint8_t, 16> animLen = {};
+	std::array<int16_t, 16> animWidth = {};
+	std::array<int16_t, 16> animWidth2 = {};
+
+	std::vector<std::unique_ptr<byte[]>> pinnedMem;

```

Comment:
Does this need to be exposed? Could be worth changing MisFileData to a class and making this private.

Reply:
You're right, pinnedMem doesn't have to be exposed.

Reply:
The rest of them do.